src/overlays/actors/ovl_En_Encount4/z_en_encount4.c

```diff
@@ -33,20 +32,173 @@ const ActorInit En_Encount4_InitVars = {
+void func_809C4078(EnEncount4* this, PlayState* play) {
+    s16 rotY;
+    Vec3f pos;
+    s32 fireWallParams;
+    s32 i;
+    EnBsb* captainKeeta = this->captainKeeta;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount4/EnEncount4_Destroy.s")
+    if ((this->switchFlags >= 0) && Flags_GetSwitch(play, this->switchFlags)) {
+        this->timer = 100;
+        this->actionFunc = func_809C464C;
+    } else if (BREG(1) == 0) {
+        if ((this->captainKeeta->actor.id != ACTOR_EN_BSB) || (captainKeeta->actor.update == NULL)) {
+            Actor_MarkForDeath(&this->actor);
+        } else if ((this->unk148 != 0) || (this->actor.xzDistToPlayer < 240.0f)) {
+            if ((this->unk148 == 0) && (captainKeeta->unk2DC != 0)) {
+                Actor_MarkForDeath(&this->actor);
+            } else {
+                fireWallParams = BGFIREWALL_PARAMS_0;
+                if ((this->unk148 == 0) || (captainKeeta->unk2DC != 0)) {
+                    i = 0;
+                    if (this->unk148 != 0) {
+                        fireWallParams = BGFIREWALL_PARAMS_1;
+                        i = 2;
+                    }
+                    while (i < ARRAY_COUNT(D_809C46DC)) {
+                        rotY = D_809C46D0[i] + this->actor.world.rot.y;
+                        pos.x = Math_SinS(rotY) * D_809C46DC[i] + this->actor.world.pos.x;
+                        pos.y = this->actor.world.pos.y;
+                        pos.z = Math_CosS(rotY) * D_809C46DC[i] + this->actor.world.pos.z;
+                        Actor_SpawnAsChild(&play->actorCtx, &this->actor, play, ACTOR_BG_FIRE_WALL, pos.x, pos.y, pos.z,
+                                           0, this->actor.world.rot.y, 0, fireWallParams);
+                        i++;
+                    }
+                    this->actionFunc = func_809C42A8;
+                }
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount4/func_809C3FD8.s")
+void func_809C42A8(EnEncount4* this, PlayState* play) {
+    Player* actor = GET_PLAYER(play);
+    EnBsb* captainKeeta = this->captainKeeta;
+    Vec3f pos;
+    f32 yIntersect;
+    s16 yRot;
+    CollisionPoly* colPoly;
+    s32 bgId;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount4/func_809C4078.s")
+    if (this->switchFlags >= 0) {
+        if (Flags_GetSwitch(play, this->switchFlags)) {
+            this->timer = 100;
+            this->actionFunc = func_809C464C;
+            return;
+        }
+    }
+    if (this->unk148 == 1) {
+        if ((this->captainKeeta->actor.id != ACTOR_EN_BSB) || (captainKeeta->actor.update == NULL)) {
+            Actor_MarkForDeath(&this->actor);
+        }
+    } else if (this->unk14E >= 2) {
+        this->timer = 100;
+        this->actionFunc = func_809C464C;
+    } else if ((gSaveContext.save.weekEventReg[85] & 0x40) || (this->unk14C >= 2) ||
+               (this->actor.xzDistToPlayer > 240.0f)) {
+        return;
+    } else {
```

Comment:
Since you have the early return, I'd say just remove the else

Reply:
This doesn't match. I can remove the return and keep the else, though.

Reply:
You sure? Could you post a scratch? I'm just wondering if you combined the if's:
`if ((this->switchFlags >= 0) && (Flags_GetSwitch(play, this->switchFlags))) {`
?